src/runtime/ObjectTemplate.cpp

```diff
@@ -345,13 +434,22 @@ bool ObjectTemplate::installTo(Context* ctx, Object* target)
     return result.error.isEmpty();
 }
 
-void ObjectTemplate::setNamedPropertyHandler(const ObjectTemplateNamedPropertyHandlerData& data)
+void ObjectTemplate::setNamedPropertyHandler(const ObjectTemplatePropertyHandlerConfiguration& data)
+{
+    ASSERT(!m_propertyHandler);
+    m_propertyHandler = new ObjectTemplatePropertyHandlerData(data, true);
+}
+
+void ObjectTemplate::setIndexedPropertyHandler(const ObjectTemplatePropertyHandlerConfiguration& data)
 {
-    m_namedPropertyHandler = new (GC) ObjectTemplateNamedPropertyHandlerData(data);
+    ASSERT(!m_propertyHandler);
+    m_propertyHandler = new ObjectTemplatePropertyHandlerData(data, false);
```

Comment:
IMHO users want to both of named handler and indexed handler
did you check usecase?

Reply:
Good catch! I checked its cases and found that both property handlers can be registered together.
So I fixed this patch to handle it and added test case to check it. Thanks!